src/com/goide/util/GoUtil.java

```diff
@@ -195,9 +195,9 @@ public static boolean couldBeReferenceTo(@NotNull PsiElement definition, @NotNul
       public Result<Collection<String>> compute() {
         Collection<String> set = ContainerUtil.newLinkedHashSet();
         for (PsiFile file : dir.getFiles()) {
-          if (file instanceof GoFile) {
+          if (allowed(file)) {
```

Comment:
Do we need to account packages in files that are not allowed for build?
https://github.com/golang/go/blob/master/src/go/build/build.go#L447
I'm not sure if I get it right.